answerking_app/models/serializers.py

```diff
@@ -66,14 +65,38 @@ def validate_description(self, value: str) -> str:
         return compress_white_spaces(value)
 
 
-class CategorySerializer(WritableNestedModelSerializer):
+class CategorySerializer(serializers.ModelSerializer):
     name = serializers.CharField(
         max_length=50,
         allow_blank=False,
         validators=[RegexValidator("^[a-zA-Z !]+$")],
     )
     items = ItemSerializer(many=True, required=False)
 
+    def create(self, validated_data: dict) -> Category:
+        items: list[Item] = self.items_check(validated_data)
+        category: Category = Category.objects.create(
+            name=validated_data["name"]
+        )
+        category.items.add(*items)
+        return category
+
+    def update(self, category: Category, validated_data: dict) -> Category:
+        items: list[Item] = self.items_check(validated_data)
+        category.name = validated_data["name"]
+        category.items.set(objs=items)
+        category.save()
+        return category
+
+    def items_check(self, validated_data: dict) -> list[Item]:
+        items: list = []
```

Comment:
should be list[Item] as this is what it will be when it is populated. Currently we could put anything in this list from a typing point of view, but actually we only want Item objects to be put in this list.